test/com/scottyplunkett/server/HTTPResponseCodeTest.java

```diff
@@ -0,0 +1,33 @@
+package com.scottyplunkett.server;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class HTTPResponseCodeTest {
+
+    @Test
+    void retrieve() {
+        assertEquals("302 Found", HTTPResponseCode.retrieve(302));
+        assertEquals("200 OK", HTTPResponseCode.retrieve(200));
+        assertEquals("206 Partial Content", HTTPResponseCode.retrieve(206));
+        assertEquals("204 No Content", HTTPResponseCode.retrieve(204));
+        assertEquals("302 Found", HTTPResponseCode.retrieve(302));
+        assertEquals("400 Bad Request", HTTPResponseCode.retrieve(400));
+        assertEquals("401 Unauthorized", HTTPResponseCode.retrieve(401));
+        assertEquals("404 Not Found", HTTPResponseCode.retrieve(404));
+        assertEquals("405 Method Not Allowed", HTTPResponseCode.retrieve(405));
+        assertEquals("416 Range Not Satisfiable", HTTPResponseCode.retrieve(416));
+        assertEquals("409 Conflict", HTTPResponseCode.retrieve(409));
+        assertEquals("418 I'm a teapot", HTTPResponseCode.retrieve(418));
+        assertEquals("500 Internal Server Error", HTTPResponseCode.retrieve(500));
+    }
+
+    @Test
+    void encode() {
+        assertEquals(302, HTTPResponseCode.encode("/redirect"));
```

Comment:
I like this.  It's pretty neat and clean looking.  It'd be nice to see it sort of interlocked with the rest of the domain somehow - something other than string literals, maybe?

Reply:
Do you mean it would be nice to see it eat another domain object, or that the string being tested would be better as a named local string variable to give a better idea of what the string represents/comes from?